internal/policy/container/has_modified_files.go

```diff
@@ -180,69 +196,242 @@ func (p HasModifiedFilesCheck) Metadata() check.Metadata {
 	}
 }
 
-// Untar takes a destination path and a reader; a tar reader loops over the tarfile
-// creating the file structure at 'dst' along the way, and writing any files
-func untar(pathChan chan<- string, r io.Reader) error {
-	tr := tar.NewReader(r)
+func extractPackageNameVersionRelease(pkgList []*rpmdb.PackageInfo) map[string]struct{} {
+	pkgNameList := make(map[string]struct{}, len(pkgList))
+	for _, pkg := range pkgList {
+		pkgNameList[fmt.Sprintf("%s-%s-%s", pkg.Name, pkg.Version, pkg.Release)] = struct{}{}
+	}
+	return pkgNameList
+}
 
-	for {
-		header, err := tr.Next()
+// findRPMDB attempts to extract a valid RPMDB from layers in the order
+// they are provided. If found is false, pkglist should
+// be disregarded as any value there will be invalid.
+func findRPMDB(ctx context.Context, layer v1.Layer) (found bool, pkglist []*rpmdb.PackageInfo) {
+	logger := logr.FromContextOrDiscard(ctx)
+	var err error
+	pkglist, err = extractRPMDB(ctx, layer)
+	if err == nil {
+		id, _ := layer.Digest()
+		logger.V(log.TRC).Info("findRPMDB found an RPM db", "layer", id)
```

Comment:
Something weird is happening with the quoting here. I'm seeing. Is `id` a struct/interface? Do you need to `id.String()`?
```
time="2023-03-07T13:37:23-06:00" level=trace msg="findRPMDB found an RPM db" layer="\"sha256:0ac9bf1c61484d864e69dc9405a28bfcd479493b343029402cb6779a4b163cf5\""
```

Reply:
Fixed.